clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -334,14 +334,15 @@ public static void sleep(long ms) {
      * Instantiate the class
      */
     public static <T> T newInstance(Class<T> c) {
+        if (c == null)
+            throw new NullPointerException();
```

Comment:
It makes a difference for code that catches `KafkaException` instead of `RuntimeException`. We'd have to check every usage to make sure it was OK.